File: cs/Basic/Server.NodeTypes/Nodes/MyComplexObjectNode.cs
// Copyright (c) Traeger Industry Components GmbH. All Rights Reserved.

namespace NodeTypes
{
    using Opc.UaFx;

    internal class MyComplexObjectNode : OpcObjectNode
    {
        #region ---------- Private fields ----------

        private MySimpleVariableNode firstNode;
        private MyComplexVariableNode secondNode;

        #endregion

        #region ---------- Public constructors ----------

        public MyComplexObjectNode(OpcName name)
            : base(name)
        {
            this.InitializeChildren();
        }

        public MyComplexObjectNode(IOpcNode parent, OpcName name)
            : base(parent, name)
        {
            this.InitializeChildren();
        }

        #endregion

        #region ---------- Public properties ----------

        public MySimpleVariableNode FirstNode
        {
            get => this.firstNode;
        }

        public MyComplexVariableNode SecondNode
        {
            get => this.secondNode;
        }

        #endregion

        #region ---------- Protected properties ----------

        protected override OpcNodeId DefaultTypeDefinitionId
        {
            get => "ns=1;s=MyComplexObjectType";
        }

        #endregion

        #region ---------- Private methods ----------

        private void InitializeChildren()
        {
            this.firstNode = new MySimpleVariableNode(
                    parent: this,
                    name: "1:First",
                    value: 10);

            this.secondNode = new MyComplexVariableNode(
                    parent: this,
                    name: "1:Second",
                    mandatoryValue: "A",
                    optionalValue: 20);
        }

        #endregion
    }
}

File: vb/Basic/Server.NodeTypes/Nodes/MyComplexObjectNode.vb
'Copyright (c) Traeger Industry Components GmbH. All Rights Reserved.

Imports Opc.UaFx

Namespace NodeTypes
    Friend Class MyComplexObjectNode
        Inherits OpcObjectNode

#Region "---------- Private fields ----------"

        Private _firstNode As MySimpleVariableNode
        Private _secondNode As MyComplexVariableNode

#End Region

#Region "---------- Public constructors ----------"

        Public Sub New(ByVal name As OpcName)
            MyBase.New(name)
            Me.InitializeChildren()
        End Sub

        Public Sub New(ByVal parent As IOpcNode, ByVal name As OpcName)
            MyBase.New(parent, name)
            Me.InitializeChildren()
        End Sub

#End Region

#Region "---------- Public properties ----------"

        Public ReadOnly Property FirstNode As MySimpleVariableNode
            Get
                Return Me._firstNode
            End Get
        End Property

        Public ReadOnly Property SecondNode As MyComplexVariableNode
            Get
                Return Me._secondNode
            End Get
        End Property

#End Region

#Region "---------- Protected properties ----------"

        Protected Overrides ReadOnly Property DefaultTypeDefinitionId As OpcNodeId
            Get
                Return "ns=1;s=MyComplexObjectType"
            End Get
        End Property

#End Region

#Region "---------- Private methods ----------"

        Private Sub InitializeChildren()
            Me._firstNode = New MySimpleVariableNode(
                    parent:=Me,
                    name:="1:First",
                    value:=10)

            Me._secondNode = New MyComplexVariableNode(
                    parent:=Me,
                    name:="1:Second",
                    mandatoryValue:="A",
                    optionalValue:=20)
        End Sub

#End Region
    End Class
End Namespace
